Event for withers and ender dragon block destruction

Fadenfire opened the pull request:
Basically an event form of Block#canEntityDestroy. Cancelling the event prevents block damage.

Commit:
Add event

Commit:
Fix typo

mezz commented:
Looks good. 
Please explain what exactly you want to use this new event for.

Fadenfire commented:
Basically I want to have to effects of mobGriefing applied to the wither and ender dragon, but not everything else like creepers and endermen.

mezz commented:
Could that be a forge config option then?

Fadenfire commented:
Probably not, no. I would like to check if the entity is within range of a certain block, if so prevent them from destroying blocks.

mezz commented:
Ok, that's not really what you said before at all.
Makes sense.

Fadenfire commented:
Sorry I'm confusing you. I want to use this in two different mods, one where you can enable and disable wither and ender dragon block destruction in the mod's config file and another that adds a block that does what I just said above.

mezz commented:
It's important to tell us all about the cool things you want to do with the new event so that adding it is justified. It's also necessary so that anyone reviewing can brainstorm to see if there is a simpler way to accomplish what you want to do. Without that, there is no way your PR will be accepted. I want either your problem to be resolved or your PR accepted, so that's why I'm harassing you to explain more. Hope that helps.

HenryLoenwind commented:
I'm quite sure* you missed the Wither's "blue head" projectile...

(* No workspace access atm)

Madsthunder commented:
@Silly511 Why not do this for Zombies and Husks also?

Fadenfire commented:
@HenryLoenwind Yeah I guess I did. I'll fix that.

@Madsthunder How do zombies and husks destroy blocks?

HenryLoenwind commented:
Zombies break down doors on difficulty hard...

Fadenfire commented:
Oh sorry, I always though they just opened them. I've never really played awhile on hard. Yeah I guess I could expand this to include zombies.

Commit:
Merge remote-tracking branch 'MinecraftForge/1.11.x' into entity-destroy-block-event

Commit:
Extend event to include zombies and wither skulls

Commit:
Fix things

# Conflicts:
#
src/main/java/net/minecraftforge/event/entity/living/LivingDestroyBlockE
vent.java

LexManos commented on patches/minecraft/net/minecraft/entity/ai/EntityAIBreakDoor.java.patch:
```diff
@@ -0,0 +1,11 @@
+             return false;
+         }
+-        else if (!this.field_75356_a.field_70170_p.func_82736_K().func_82766_b("mobGriefing"))
++        else if (!this.field_75356_a.field_70170_p.func_82736_K().func_82766_b("mobGriefing") || !net.minecraftforge.event.ForgeEventFactory.onEntityDestroyBlock(this.field_75356_a, this.field_179507_b, this.field_75356_a.field_70170_p.func_180495_p(this.field_179507_b)))
```
worth adding a canEntityDestroy check in here?

Fadenfire replied:
Sure.

Commit:
Merge remote-tracking branch 'MinecraftForge/1.11.x' into entity-destroy-block-event

Commit:
Add a canEntityDestroy check to EntityAIBreakDoor

Commit:
Merge remote-tracking branch 'MinecraftForge/1.11.x' into entity-destroy-block-event
